st2common/st2common/config.py

```diff
@@ -388,6 +388,10 @@ def register_opts(ignore_errors=False):
         cfg.BoolOpt(
             'enable', default=True,
             help='Whether to enable or disable the ability to post a trigger on action.'),
+        cfg.ListOpt(
+            'emit_when', default=[],
```

Comment:
For explicitness sake, you could perhaps default this option to `LIVEACTION_COMPLETED_STATES`.

This way you could also simple the code here a bit - https://github.com/StackStorm/st2/pull/4315/files#diff-70b51c2693ac263742fc28ad30ebc932R234.

Reply:
> For explicitness sake, you could perhaps default this option to LIVEACTION_COMPLETED_STATES.
If we specify the list here explicitly, then we need to follow `LIVEACTION_COMPLETED_STATES` changes that may happen in the future. So I thought this way is better but I'm OK with either option. What do you think?

> This way you could also simple the code here a bit
Anyways that test case still needs to be there because users might have chance to specify a null or a blank list.

Reply:
I will merge this into master and open a small PR with changelog entry and this change here which will allow us to slightly simplify the if statement.

Reply:
And another reason why I prefer to use a default value here is because it's more explicit.

If this option contain no default value here, user needs to dig through the code and documentation to figure out what a default value of `None` indicates.